src/ir/comp.rs

```diff
@@ -481,6 +486,9 @@ impl CompInfo {
         debug!("CompInfo::from_ty({:?}, {:?})", kind, cursor);
 
         let mut ci = CompInfo::new(kind);
+        ci.is_forward_declaration = location.map_or(true, |cur|
+                                        (cur.kind() == CXCursor_StructDecl || cur.kind() == CXCursor_UnionDecl) &&
```

Comment:
@emilio It might be a good idea to add a test for unions as well, since none of the existing tests are testing it.